fdbserver/storageserver.actor.cpp

```diff
@@ -3273,6 +3274,9 @@ ACTOR Future<bool> asyncPrepareVersionsForCommit_impl(StorageServerDisk* self, S
 			}
 			if (stopEarly.isReady()) {
 				// Previous commit is done.
+				if (durable.isError()) {
```

Comment:
Technically `durableMinDelay` or some other future joined into `stopEarly` after some logic change later could also produce an error that we would want to surface, so I think it's more proper and future proof (pun intended) to check for `stopEarly.isError()` and throw its error if there is one.

Reply:
Done. I briefly took a look at the delay code and didn't realize it could throw an error.

Reply:
Sure if you consider the code as written including the caller, the delay future would not currently throw.  But what I'm saying is that both `durableMinDelay` and `durable` are just parameters to this function so from the perspective of `asyncPrepareVersionsForCommit_impl()` they are just futures to interact with, could have errors, and those errors should be thrown.

Reply:
The change made is not what I intended.  This is what I am saying to do:

```
if (stopEarly.isReady()) {
  if(stopEarly.isError()) {
    throw stopEarly.getError();
  }
}
```

This will surface whatever error caused `stopEarly` to become ready.  There is no need to know or consider what the components of `stopEarly` are and check them individually.

FWIW, two shorter but less clear equivalents of this would be:

```
if (stopEarly.isReady()) {
  stopEarly.get();
}
```

or

```
if (stopEarly.isReady()) {
  wait(stopEarly);
}
```